### PR TITLE
[test] Separate out the network tests

### PR DESCRIPTION
**Pull request by aravindhp**
Move the network tests out from the creation tests. This is in preparation for separating out the upgrade and reconfiguration tests into a separate CI job.

### CODE REVIEW

**Comment by aravindhp** [comment]
/retitle [test] Separate out the network tests

File: hack/run-ci-e2e-test.sh
Comment (mansikulkarni96) [inline hack/run-ci-e2e-test.sh:130]: I see the timeout being decreased by 10 minutes in total is that on purpose, if yes is there an evidence for doing so?

File: test/e2e/network_test.go
Comment (mansikulkarni96) [inline test/e2e/network_test.go:27]: shouldn't we check for error?

**Review by mansikulkarni96 — COMMENTED** [review]
Thanks for working on this @aravindhp , PTAL at my comments.

File: hack/run-ci-e2e-test.sh
Comment (aravindhp) [reply hack/run-ci-e2e-test.sh:130]: Yes. On an average the networking e2e tests is taking around 10 minutes to complete across all providers.

File: test/e2e/network_test.go
Comment (aravindhp) [reply test/e2e/network_test.go:27]: Good catch.

File: test/e2e/network_test.go
Comment (sebsoto) [inline test/e2e/network_test.go:30]: no creation happening here, `timed out waiting for Windows nodes` is more proper

File: test/e2e/metrics_test.go
Comment (sebsoto) [inline test/e2e/metrics_test.go:31]: Why is this change being made? Is it because the metrics tests now run before the networking tests when run in CI? This will cause issues if the network tests are run independently. Instead the workload namespace creation/deletion can be handled in the common `TestWMCO` function.

**Review by sebsoto — CHANGES_REQUESTED** [review]

File: test/e2e/metrics_test.go
Comment (aravindhp) [reply test/e2e/metrics_test.go:31]: >Is it because the metrics tests now run before the networking tests when run in CI?

Yes

> Instead the workload namespace creation/deletion can be handled in the common TestWMCO function.

Moving the creation is not straightforward as `testCtx.createNamespace` would need to become `testCtx.EnsureNamespace`. Given how rarely we run the network tests independently I decided to put this off. But I can make the change as part of this PR.

File: test/e2e/network_test.go
Comment (mansikulkarni96) [reply test/e2e/network_test.go:31]: this comments need to be addressed, LGTM apart from it.

File: test/e2e/wmco_test.go
Comment (sebsoto) [inline test/e2e/wmco_test.go:36]: Doesnt this need to be cleaned up?

**Review by sebsoto — APPROVED** [review]

**Comment by sebsoto** [comment]
/hold
Approved but holding due to my comment

File: test/e2e/wmco_test.go
Comment (aravindhp) [reply test/e2e/wmco_test.go:36]: That [happens](https://github.com/openshift/windows-machine-config-operator/blob/master/test/e2e/delete_test.go#L69) in the deletion tests

**Comment by aravindhp** [comment]
/hold cancel
/retest

https://github.com/openshift/installer/pull/4779 has merged

**Pull request merged**
